src/compositor/windowpixmapitem.cpp

```diff
@@ -587,13 +604,22 @@ QSGNode *WindowPixmapItem::updatePaintNode(QSGNode *oldNode, UpdatePaintNodeData
             delete m_unmapLock;
             m_unmapLock = 0;
             prov->program->disableAttributeArray(prov->vertexLocation);
+
+            m_haveSnapshot = true;
```

Comment:
m_haveSnapshot should probably be reset to false at the start of updatePaintNode(), otherwise it will be true until the window id is changed, even if there is now a valid buffer from the client.